Improve schema to have same return as tcfapi

thcolin opened the pull request:
Hi,

I'm working on a project where I am using `tc-string-parse` beside [`@iabtcf/cmpapi`](https://github.com/InteractiveAdvertisingBureau/iabtcf-es/tree/master/modules/cmpapi). I found `tcData` returned after decoding a `tcString` doesn't follow the same schema, so here's some changes to fit to `@iabtcf/cmpapi` schema.

I added a `parent` key to some parts of the schema and `applySchemaValue` function to avoid rewrite 3x the behavior. I renamed some keys too, to fit `@iabtcf/cmpapi` schema.

This is a breaking change.

thcolin committed:
Fix invalid string failing test

thcolin committed:
Improve schema to have same return as iab tcData

NikolayFrantsev commented:
Thanks for your request.
I definitely like that idea and implementation is almost perfect.
Could you please add extra test to compare library response format with model from iabcmp?

NikolayFrantsev commented on tc-string-parse.js:
```diff
@@ -79,6 +79,28 @@
+                }
+
+                if (typeof result[schema.parent][schema.key] === "object" && typeof value === "object") {
+                    result[schema.parent][schema.key] = Object.assign({}, result[schema.parent][schema.key], value);
```
`Object.assign` is not supported in IE. Could you please extract [assign logic](https://github.com/audienceproject/tc-string-parse/blob/7fdfd5875f14e3fad0d4ff7eaee247bdc4e1888a/tc-string-parse.js#L288-L289) into additional function and use that?

thcolin commented:
Hi,

I added a test to check structure with `tcData.fixture.json`. I was unable to use both iab libraries to decode a string in test environment:
* `@iabtcf/cmpapi`: use `window`, too complex to handle in test environment
* `@iabtcf/core`: doesn't return end structure

I've added a `objectAssign` function to polyfill behavior as requested to be compatible with IE.

NikolayFrantsev commented:
I'll add some extra changes and publish new release next week.
Thanks for your work.

thcolin commented:
Thanks for your time and interest !
